<launch>
    <arg name="demo" default="True" description="bool: true will launch the Franka demo.launch.py. False will open rviz, but not the demo"/>

    <include file="$(find-pkg-share franka_fer_moveit_config)/launch/demo.launch.py" if="$(eval $(var demo)==True)"/>
    <include file="$(find-pkg-share franka_fer_moveit_config)/launch/moveit_rviz.launch.py" if="$(eval $(var demo)==False)">
        <arg  name ="robot_ip" value="panda0.robot"/>
    </include>

    <!--Launch the Realsense Camera Node (bug_god)-->
    <include file="$(find-pkg-share realsense2_camera)/launch/rs_launch.py">
        <arg name="camera_name" value="bug_god"/>
        <arg name="camera_namespace" value="camera"/>
        <arg name="serial_no" value="'139522074035'"/>
        <arg name="pointcloud.enable" value="false"/>
        <arg name="color_width" value="640"/>
        <arg name="color_height" value="480"/>
        <arg name="color_fps" value="15"/>
        <arg name="depth_fps" value="15"/>
    </include>

    <!--Static Transform Publisher from hand to new_tcp-->
    <!-- <node pkg="tf2_ros" exec="static_transform_publisher" name="new_tcp_tf_publisher"
      args="0.1 0.0 0.13 0.0 0.0 0.0 fer_hand_tcp new_tcp" /> -->

    <!--Launch the April tag identification Node-->
    <node pkg="apriltag_ros" exec="apriltag_node" name="apriltag">
        <!-- Remaps to RealSense topics -->
        <remap from="image_rect" to="/camera/bug_god/color/image_raw"/>
        <remap from="camera_info" to="/camera/bug_god/color/camera_info"/>
        <param from = "$(find-pkg-share apriltag_ros)/cfg/tags_16h5.yaml"/>
    </node>

    <!--Load the Target Decision Node-->
    <node pkg="bug_catcher" exec="target_decision_node" name="target_decision_node" output="screen">
        <param name="base_frame" value="base"/>
        <param name="default_color" value="blue"/>
        <param name="camera_height" value="0.5"/>
        <param from = "$(find-pkg-share bug_catcher)/config/calibration_parameters.yaml"/>
        <param name="color_path_sky_cam" value="$(find-pkg-share bug_catcher)/config/sky_cam_colors.yaml"/>
    </node>

    <!--Load the Sort Node-->
    <node pkg="bug_catcher" exec="sort_node" name="sort_node" output="screen">
        <param name="filename" value="$(find-pkg-share bug_catcher)/config/objects.yaml"/>
    </node>

    <!--RQT Image Viewer-->
    <node pkg="rqt_image_view" exec="rqt_image_view" name="wrist_camera_debug_view"
          args="/wrist_camera/debug_view"/>
    <node pkg="rqt_image_view" exec="rqt_image_view" name="bug_god_debug_view"
          args="/bug_god/debug_view"/>

    <!-- Needed for the picking and moving of still bugs -->
    <!-- <node pkg="bug_catcher" exec="pick_node">
        <param name="filename" value="$(find-pkg-share bug_catcher)/config/objects.yaml"/>
    </node> -->


</launch>